client/client.go

```diff
@@ -12,27 +12,37 @@ import (
 	"net/url"
 	"os"
 	"regexp"
+	"strings"
 	"time"
 )
 
 // NewClient returns a new HTTP client for accessing
 // Create/List/Get/Cancel Task endpoints. "address" is the address
 // of the TES server.
-func NewClient(address string) *Client {
+func NewClient(address string) (*Client, error) {
 	password := os.Getenv("FUNNEL_SERVER_PASSWORD")
 
-	// Strip trailing slash. A quick and dirty fix.
-	re := regexp.MustCompile("/+$")
-	address = re.ReplaceAllString(address, "")
+	re := regexp.MustCompile("^(.+://)?(.[^/]+)(.+)?$")
+	endpoint := re.ReplaceAllString(address, "$1$2")
+
+	reScheme := regexp.MustCompile("^.+://")
+	switch reScheme.MatchString(endpoint) {
+	case true:
```

Comment:
`switch` for true/false is pretty weird. `if` is better.

Reply:
Good point.